Modules/Core/SpatialObjects/include/itkImageMaskSpatialObject.h

```diff
@@ -71,6 +75,23 @@ class ITK_TEMPLATE_EXPORT ImageMaskSpatialObject:
   bool IsInsideInObjectSpace(const PointType & point, unsigned int depth=0,
     const std::string & name="") const override;
 
+#if ! defined ( ITK_LEGACY_REMOVE )
+  itkLegacyMacro( typename ImageType::RegionType GetAxisAlignedBoundingBoxRegion() )
+  {
+    using LabelType = unsigned char;
+    using ShapeLabelObjectType = itk::ShapeLabelObject< LabelType, TDimension >;
+    using LabelMapType = itk::LabelMap< ShapeLabelObjectType >;
+
+    using LI2SLMFType = itk::LabelImageToShapeLabelMapFilter<ImageType,LabelMapType >;
+    typename LI2SLMFType::Pointer i2l = LI2SLMFType::New();
+    i2l->SetInput(this->GetImage());
+    i2l->Update();
+    LabelMapType *labelMap = i2l->GetOutput();
+    ShapeLabelObjectType *labelObject = labelMap->GetNthLabelObject(1);
+    return labelObject->GetBoundingBox();
+  }
+#endif
+
```

Comment:
Just trying to understand the code. Do I understand correctly that the pixel values are always either 0 or 1? And is there always an `labelMap->GetNthLabelObject(1)`, even when the image is entirely black? Or when it is uniformly filled with 1?

Reply:
I think only `GetNthLabelObject(0)` is guaranteed to exist.

Reply:
I'm removing this override.  It does not work in a generic way.  We need to find a generic way to express the old behavior, even if it is just for documentation.  That function was / is a HUGE burden to replace in the code base for tools that depend on it (i.e. ANTs/ BRAINSTools /  Slicer by extension).

If @N-Dekker has an alterant algorithm to produce the previous results, I would welcome it with wide open arms.

Reply:
@hjmjohnson Here is my implementation `EstimateAxisAlignedBoundingBoxRegion`:  https://gist.github.com/N-Dekker/9e1f5bcc6f65c7652c11d1cbd3d519ee

It appears to work well. However, I'm still wondering, why didn't you simply place the original `GetAxisAlignedBoundingBoxRegion()` implementation with an `#if !defined( ITK_LEGACY_REMOVE )' block?

Reply:
@N-Dekker Because @aylward in the migration guide suggests using the Label map approach.

Reply:
I restored the previous interface for GetAxisAlignedBoundingBoxRegion verbatim for what it used to be.  I'll leave an improved interface integration upto @N-Dekker to deal with new features.